paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerQuitsScriptEventPaperImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.player.PlayerQuitsScriptEvent;
+import com.denizenscript.denizen.paper.PaperModule;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import net.md_5.bungee.api.ChatColor;
+
+
+public class PlayerQuitsScriptEventPaperImpl extends PlayerQuitsScriptEvent {
+
+    public PlayerQuitsScriptEventPaperImpl() {
+        registerSwitches("cause");
+        this.<PlayerQuitsScriptEventPaperImpl>registerTextDetermination("none", (evt) -> {
+            event.quitMessage(null);
+        });
+        this.<PlayerQuitsScriptEventPaperImpl, ElementTag>registerOptionalDetermination(null, ElementTag.class, (evt, context, determination) -> {
+            event.quitMessage(PaperModule.parseFormattedText(determination.toString(), ChatColor.WHITE));
+            return true;
+        });
```

Comment:
Don't think this needs a `registerOptionalDetermination`? but also not sure if double-registering determinations like that is great - can wait to see what monkey says, but might be better to have `get/setQuitMessage` methods that are used for the determination/context in the default impl, and then this impl can just override them.

Reply:
Needs to be fixed in this override as well